src/main/java/io/supertokens/inmemorydb/queries/TOTPQueries.java

```diff
@@ -0,0 +1,179 @@
+package io.supertokens.inmemorydb.queries;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.supertokens.inmemorydb.Start;
+import io.supertokens.inmemorydb.config.Config;
+import io.supertokens.pluginInterface.RowMapper;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.TOTPUsedCode;
+
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.execute;
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.update;
+
+public class TOTPQueries {
+    public static String getQueryToCreateUserDevicesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getTotpUserDevicesTable() + " ("
+                + "user_id VARCHAR(128) NOT NULL," + "device_name VARCHAR(256) NOT NULL,"
+                + "secret_key VARCHAR(256) NOT NULL,"
+                + "period INTEGER NOT NULL," + "skew INTEGER NOT NULL," + "verified BOOLEAN NOT NULL,"
+                + "PRIMARY KEY (user_id, device_name))";
+    }
+
+    public static String getQueryToCreateUsedCodesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getTotpUsedCodesTable() + " ("
+                + "user_id VARCHAR(128) NOT NULL," + "code CHAR(6) NOT NULL," + "is_valid_code BOOLEAN NOT NULL,"
+                + "expiry_time_ms BIGINT UNSIGNED NOT NULL,"
+                + "FOREIGN KEY (user_id) REFERENCES " + Config.getConfig(start).getTotpUserDevicesTable()
+                + "(user_id) ON DELETE CASCADE)";
```

Comment:
```suggestion
                + "FOREIGN KEY (user_id, device_name) REFERENCES " + Config.getConfig(start).getTotpUserDevicesTable()
                + "(user_id, device_name) ON DELETE CASCADE)";
```

Reply:
Not using this anymore. So not relevant.